prog/dftb+/lib_extlibs/negf_int.F90

```diff
@@ -1360,24 +1366,32 @@ subroutine calc_current(groupKS, ham, over, iNeighbor, nNeighbor, iAtomStart, iP
     real(dp), pointer    :: currPMat(:,:)=>null()
     real(dp), pointer    :: ldosPMat(:,:)=>null()
     real(dp), pointer    :: currPVec(:)=>null()
-    integer :: iKS, iK, iS, nKS, ii, err, ncont, readSGFbkup
+    integer :: iKS, iK, iS, nKS, nS,  nTotKS, ii, err, ncont, readSGFbkup
     type(unit) :: unitOfEnergy        ! Set the units of H
     type(unit) :: unitOfCurrent       ! Set desired units for Jel
     type(lnParams) :: params
 
     integer :: i, j, k, NumStates, icont
     real(dp), dimension(:,:), allocatable :: H_all, S_all
     character(:), allocatable :: filename
+    character(2) :: id1, id2
 
 #:if WITH_MPI
-    call negf_mpi_init(mpicomm)
+    call negf_mpi_init(enecomm, tIOproc)
 #:endif
     call get_params(negf, params)
 
     unitOfEnergy%name = "H"
     unitOfCurrent%name = "A"
 
+    ! groupKS is local, hence nKS il local
     nKS = size(groupKS, dim=2)
+#:if WITH_MPI
+    call mpifx_allreduce(kscomm, nKS, nTotKS, MPI_SUM)
+#:else
+    nTotKS = nKS
+#:endif
+    nS = nTotKS/size(kpoints, 2)
```

Comment:
Is there any reason not to use nS = size(ham,dim=2); nTotKS = nS * size(kpoints, dim=2)? This would remove the need  for the reduce operation.

Reply:
Yes. I think it is easier. Provided I add a check: nS=size(ham,2); if (nS>2)  nS=1